format/spec.md

```diff
@@ -841,14 +842,38 @@ Notes:
+* Position delete files identify deleted rows by file location and row position
+* Equality delete files identify deleted rows by the value of one or more columns
 
-Row-level delete files are tracked by manifests, like data files. A separate set of manifests is used for delete files, but the manifest schemas are identical.
+Deletion vectors are a binary representation of deletes for a single data file that is more efficient at execution time than position delete files. Unlike equality or position delete files, there can be at most one deletion vector for a given data file in a table. Writers must detect concurrently added deletion vectors and merge multiple vectors for a given data file before committing.
 
-Both position and equality deletes allow encoding deleted row values with a delete. This can be used to reconstruct a stream of changes to a table.
+Row-level delete files (both equality and position delete files) are valid Iceberg data files: files must use valid Iceberg formats, schemas, and column projection. It is recommended that these delete files are written using the table's default file format.
 
+Row-level delete files and deletion vectors are tracked by manifests. A separate set of manifests is used for delete files and DVs, but the same manifest schema is used for both data and delete manifests. Deletion vectors are tracked individually by file location, offset, and length within the containing file. Deletion vector metadata must include the referenced data file.
+
+Both position and equality delete files allow encoding deleted row values with a delete. This can be used to reconstruct a stream of changes to a table.
+
+
+### Deletion Vectors
+
+Delete vectors identify deleted rows of a file by encoding deleted positions in a bitmap. A set bit at position P indicates that the row at position P is deleted.
+
+Deletion vectors are stored using the `delete-vector-v1` blob definition from the [Puffin spec][puffin-spec].
+
+Deletion vectors support positive 64-bit positions, but are optimized for cases where most positions fit in 32 bits by using a collection of 32-bit Roaring bitmaps. 64-bit positions are divided into a 32-bit "key" using the most significant 4 bytes and a 32-bit position using the least significant 4 bytes. For each key in the set of positions, a 32-bit Roaring bitmap is maintained to store a set of 32-bit positions for that key.
+
+To test whether a certain position is set, its most significant 4 bytes (the key) are used to find a 32-bit bitmap and the least significant 4 bytes are tested for inclusion in the bitmap. If a bitmap is not found for the key, then it is not set.
+
+Delete manifests track deletion vectors individually by the containing file location, starting offset of the DV magic bytes, and total length of the deletion vector blob. Multiple deletion vectors can stored in the same file. There are no restrictions on the data files that can be referenced by deletion vectors in the same file.
```

Comment:
```suggestion
Delete manifests track deletion vectors individually by the containing file location, starting offset of the DV magic bytes, and total length of the deletion vector blob. Multiple deletion vectors can stored in the same file. There are no restrictions on which data files can be referenced by deletion vectors in the same puffin file.
```

Reply:
I think this should use `that` instead of `which` because "no restrictions on which" is unclear. The phrase "on which" typically refers to the preceding noun, as in the phrase "restrictions on which atomicity depends".

I'll update to call out these are "Puffin" files.